memgpt/client/client.py

```diff
@@ -934,22 +959,41 @@ def get_tool_id(self, name: str) -> str:
     def load_data(self, connector: DataConnector, source_name: str):
         self.server.load_data(user_id=self.user_id, connector=connector, source_name=source_name)
 
-    def create_source(self, name: str):
-        return self.server.create_source(user_id=self.user_id, name=name)
+    def create_source(self, name: str) -> Source:
+        request = SourceCreate(name=name)
+        return self.server.create_source(request=request, user_id=self.user_id)
 
-    def delete_source(self, source_id: Optional[uuid.UUID] = None, source_name: Optional[str] = None):
+    def delete_source(self, source_id: str):
         # TODO: delete source data
-        self.server.delete_source(user_id=self.user.id, source_id=source_id, source_name=source_name)
+        self.server.delete_source(source_id=source_id, user_id=self.user_id)
+
+    def get_source(self, source_id: str) -> Source:
+        return self.server.get_source(source_id=source_id, user_id=self.user_id)
+
+    def get_source_id(self, source_name: str) -> str:
+        return self.server.get_source_id(source_name=source_name, user_id=self.user_id)
 
-    def get_source(self, source_id: Optional[uuid.UUID] = None, source_name: Optional[str] = None):
-        return self.server.ms.get_source(user_id=self.user_id, source_id=source_id, source_name=source_name)
+    def attach_source_to_agent(self, source_id: str, agent_id: str):
+        request = SourceAttach(agent_id=agent_id, source_id=source_id)
+        self.server.attach_source_to_agent(request=request, user_id=self.user_id)
 
-    def attach_source_to_agent(self, source_id: uuid.UUID, agent_id: uuid.UUID):
-        self.server.attach_source_to_agent(user_id=self.user_id, source_id=source_id, agent_id=agent_id)
+    def detach_source_from_agent(self, source_id: str, agent_id: str):
+        request = SourceDetach(agent_id=agent_id, source_id=source_id)
+        self.server.detach_source_from_agent(request=request, user_id=self.user_id)
 
-    def list_sources(self):
+    def list_sources(self) -> List[Source]:
         return self.server.list_all_sources(user_id=self.user_id)
 
+    def list_attached_sources(self, agent_id: str) -> List[Source]:
+        return self.server.list_attached_sources(agent_id=agent_id)
+
+    def update_source(self, source_id: str, name: Optional[str] = None) -> Source:
+        # TODO should the arg here just be "source_update: Source"?
```

Comment:
@sarahwooders question here

If answer is "yes", then shouldn't we remove `SourceUpdate` from the schemas?